spring-context/src/main/java/org/springframework/scheduling/annotation/AbstractAsyncConfiguration.java

```diff
@@ -62,20 +61,12 @@ public void setImportMetadata(AnnotationMetadata importMetadata) {
 		}
 	}
 
-	/**
-	 * Collect any {@link AsyncConfigurer} beans through autowiring.
-	 */
-	@Autowired(required = false)
-	void setConfigurers(Collection<AsyncConfigurer> configurers) {
-		if (CollectionUtils.isEmpty(configurers)) {
-			return;
-		}
-		if (configurers.size() > 1) {
-			throw new IllegalStateException("Only one AsyncConfigurer may exist");
-		}
-		AsyncConfigurer configurer = configurers.iterator().next();
-		this.executor = configurer::getAsyncExecutor;
-		this.exceptionHandler = configurer::getAsyncUncaughtExceptionHandler;
+	@Autowired
+	void setConfigurers(Optional<AsyncConfigurer> configurerOptional) {
+		configurerOptional.ifPresent(configurer -> {
```

Comment:
This could throw `NullPointerException` as `configurerOptional` can be null. In addition, `Optional` is rather best used for showing that a computation might not return a result than an optional input parameter. For this case, static polymorphism (overloading) would be a better option

Reply:
`configurerOptional` can't be null. Spring does not allow to inject `null Optional` - Optional was invented to avoid null problems.

Anyway, I modified signature and removed `Optional` - but for simplicity, not for null fear

Reply:
That's correct, Spring doesn't allow `null` injection. But leaving injection aside, that is still a function that can be invoked by some other class from the `org.springframework.scheduling.annotation` package (although is not the case for now)